gson/src/main/java/com/google/gson/ReflectionAccessFilter.java

```diff
@@ -0,0 +1,187 @@
+     * the used Java version would only log a warning, or is configured to open
+     * packages for reflection using command line arguments.
+     *
+     * @see AccessibleObject#setAccessible(boolean)
+     * @see AccessibleObject#canAccess(Object)
+     */
+    BLOCK_INACCESSIBLE,
+    /**
+     * Blocks all reflection access for the class. Other means for serializing
+     * and deserializing the class, such as a {@link TypeAdapter}, have to
+     * be used.
+     */
+    BLOCK_ALL
+  }
+
+  /**
+   * Blocks all reflection access to members of standard Java classes which are
+   * not accessible by default. However, reflection access is still allowed for
+   * classes for which all fields are accessible and which have an accessible
+   * no-args constructor (or for which an {@link InstanceCreator} has been registered).
+   *
+   * <p>If this filter encounters a class other than a standard Java class it
+   * returns {@link FilterResult#INDECISIVE}.
+   *
+   * <p>This filter is mainly intended to help enforcing the access checks of
+   * Java Platform Module System. It allows detecting illegal access, even if
+   * the used Java version would only log a warning, or is configured to open
+   * packages for reflection. However, this filter <b>only works for Java 9 and
+   * higher</b>, when using an older Java version its functionality will be
+   * limited.
+   *
+   * <p>Note that this filter might not cover all standard Java classes. The set
+   * of detected classes might be expanded in the future without prior notice.
+   *
+   * @see FilterResult#BLOCK_INACCESSIBLE
+   */
+  ReflectionAccessFilter BLOCK_INACCESSIBLE_JAVA = new ReflectionAccessFilter() {
+    @Override public FilterResult check(Class<?> rawClass) {
+      return ReflectionAccessFilterHelper.isJavaType(rawClass)
+        ? FilterResult.BLOCK_INACCESSIBLE
+        : FilterResult.INDECISIVE;
+    }
+  };
+
+  /**
+   * Blocks all reflection access to members of standard Java classes.
+   *
+   * <p>If this filter encounters a class other than a standard Java class it
+   * returns {@link FilterResult#INDECISIVE}.
+   *
+   * <p>This filter is mainly intended to prevent depending on implementation
+   * details of the Java platform and to help applications prepare for upgrading
+   * to the Java Platform Module System.
+   *
+   * <p>Note that this filter might not cover all standard Java classes. The set
+   * of detected classes might be expanded in the future without prior notice.
+   *
+   * @see #BLOCK_INACCESSIBLE_JAVA
+   * @see FilterResult#BLOCK_ALL
+   */
+  ReflectionAccessFilter BLOCK_ALL_JAVA = new ReflectionAccessFilter() {
+    @Override public FilterResult check(Class<?> rawClass) {
+      return ReflectionAccessFilterHelper.isJavaType(rawClass)
+        ? FilterResult.BLOCK_ALL
+        : FilterResult.INDECISIVE;
+    }
+  };
+
+  /**
+   * Blocks all reflection access to members of standard Android classes.
```

Comment:
"A standard Android class is any class in a {@code android.*} or {@code androidx.*} package, or any standard Java class ({@code java.*} or {@code javax.*})."